File: Level-2/19.java
class Solution {
    public int[][] solution(int[][] arr1, int[][] arr2) {
        int[][] answer = new int[arr1.length][arr2[0].length];
        for(int i = 0; i < answer.length; i++){
            for(int j = 0; j < answer[i].length; j++){
                int count = 0;
                for(int x = 0; x < arr1[0].length; x++){
                    count += arr1[i][x] * arr2[x][j];
                }
                answer[i][j] = count;
            }
        }
        return answer;
    }
}

File: Level-2/19.js
function solution(n) {
    let answer = 0;
    
    const dfs = (board, row) => {
        if(row === n) answer++;
        else {
            for(let i = 1; i <= n; i++) {
                board[row+1] = i;
                if(isValid(board, row+1)) dfs(board, row+1);
            }
        }
    };
    
    const isValid = (board, row) => {
        for(let i = 1; i < row; i++) {
            if(board[i] === board[row]) return false;
            if(Math.abs(i - row) === Math.abs(board[i] - board[row])) return false;
        }
        
        return true;
    };
    
    for(let i = 1; i <= n; i++) {
        const board = Array.from(n+1).fill(0);
        board[1] = i;
        dfs(board, 1);
    }
    
    return answer;
}
